### ISSUE TITLE
separate.insertPrefix may be converted from empty string to underscore

### ISSUE THREAD

**Issue by TonyGravagno** [araera111/mysql-to-zod#206]
I'm not 100% positive that this issue is in core. It might just be in my code base. But I'm reporting it here just in case, and I will look at it as time permits...

In the config file, when separate.isSeparate=true, and separate.insertPrefix is an empty string, the const name for the Insert schema is correct, but the type name for the Insert schema seems to have an underscore pre-pended (added in front) of the type name. But if separate.insertPrefix is not empty, the name of the type for the Insert schema is correct.

Example:

tablename: abc  
schema based on config: Abc_Schema  
type based on config: Abc_Type  
insert schema: Abc_New_Schema  
insert type: _Abc_New_Type ...underscore prefix

If separate.insertPrefix is set to Foo, insert type is FooAbc_New_Type.

In summary, we need to check to see how the inserPrefix is processed if it is an empty string.

I have extensive replacements, and some changes in my codebase. I'll test and look at this more - just be aware that you might see this in your own schema output.

**Comment by araera111** [comment]
Hello.

In version 2.3.1, when `isSeparate: true` and `insertPrefix: ""` are set, the same schema is generated for both select and insert in my environment. This behavior seems like a bug in itself. Should the process perhaps be interrupted when `isSeparate: true` and `insertPrefix: ""` are used together?

Below are the configuration and the schema that were generated in my environment.

```js:mysqlToZod.config.js
/** @type {import("./src/options/options").MysqlToZodOption} */
const options = {
	output: {
		outDir: "./mysqlToZod",
		fileName: "schema.ts",
		globalSchemaFileName: "globalSchema.ts",
		sqlFileName: "tables.sql",
		saveSql: true,
	},
	comments: {
		table: {
			active: true,
			format: "// [table:!name] : !text",
		},
		column: {
			active: true,
			format: "// !name : !text",
		},
	},
	type: {
		declared: "type",
		format: "pascal",
		prefix: "",
		suffix: "",
		replacements: [],
	},
	schema: {
		format: "camel",
		prefix: "",
		suffix: "Schema",
		replacements: [],
		nullType: "nullish",
		inline: false,
		zod: {
			implementation: [],
			references: [],
		},
	},
	separate: {
		isSeparate: true,
		insertPrefix: "",
		insertSuffix: "",
	},
	dbConnection: {
		database: "my_todo",
		host: "localhost",
		password: "root",
		port: 3306,
		user: "root",
	},
};
module.exports = options;
```

```ts:schema.ts
import { z } from "zod";
import { globalSchema } from "./globalSchema";
export const todoSchema = z.object({
	id: globalSchema.mysqlBIGINT,
	title: globalSchema.mysqlVARCHAR,
	description: globalSchema.mysqlTEXT.nullish(),
	is_completed: globalSchema.mysqlTINYINT.nullish(),
	created_at: globalSchema.mysqlTIMESTAMP.nullish(),
	updated_at: globalSchema.mysqlTIMESTAMP.nullish(),
});
export type Todo = z.infer<typeof todoSchema>;
export const todoSchema = z.object({
	id: globalSchema.mysqlBIGINT.nullish(),
	title: globalSchema.mysqlVARCHAR,
	description: globalSchema.mysqlTEXT.nullish(),
	is_completed: globalSchema.mysqlTINYINT.nullish(),
	created_at: globalSchema.mysqlTIMESTAMP.nullish(),
	updated_at: globalSchema.mysqlTIMESTAMP.nullish(),
});
export type Todo = z.infer<typeof todoSchema>;
```

```sql:tables.sql
CREATE TABLE `todo` (
  `id` bigint unsigned NOT NULL AUTO_INCREMENT,
  `title` varchar(255) NOT NULL,
  `description` text,
  `is_completed` tinyint (1) DEFAULT '0',
  `created_at` timestamp NULL DEFAULT CURRENT_TIMESTAMP,
  `updated_at` timestamp NULL DEFAULT CURRENT_TIMESTAMP ON UPDATE CURRENT_TIMESTAMP,
  PRIMARY KEY (`id`),
  UNIQUE KEY `id` (`id`)
) ENGINE = InnoDB DEFAULT CHARSET = utf8mb4 COLLATE = utf8mb4_0900_ai_ci
```
If you could provide the specific configuration and SQL that are not working as expected on your end, I would be happy to investigate further.

Thank you for your time and assistance.
